recipe/deploy/writable.php

```diff
@@ -23,8 +23,10 @@
     }
 
     if ($httpUser === false && ! in_array($mode, ['chgrp', 'chmod'], true)) {
-        // Detect http user in process list.
-        $httpUser = run("ps axo comm,user | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | head -1 | awk '{print $2}'");
+        // Attempt to detect http user in process list.
+        if (test("[ $(ps axo comm,user | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | sort | uniq | wc -l) -eq 1 ]")) {
+            $httpUser = run("ps axo comm,user | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | sort | uniq | head -1 | awk '{print $2}'");
```

Comment:
What about using php here? Remove `head -1` get all lines and check on php side.

Reply:
In that case, I have two suggestions
The first is 
```php
$httpUserCandidates = explode(',', run("ps axo comm,user | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | sort | uniq | awk '{print $2}' | paste -sd ',' -"));
if (count($httpUserCandidates) === 1) {
    $httpUser = $httpUserCandidates[0];
}
```
This way I don't really need to know what character is the line delimiter (\n, \r or \n\r) on the remote machine, since `paste` will solve this for me and produce comma delimited values.

But maybe this is overthinking things and the simpler approach is better
```php
$httpUserCandidates = explode("\n", run("ps axo comm,user | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | sort | uniq | awk '{print $2}'"));
if (count($httpUserCandidates) === 1) {
    $httpUser = $httpUserCandidates[0];
}
```

What would you suggest to go with?

Reply:
Hey @antonmedv, I decided to go with the second snippet. If you think using `paste` is a better option, I can revert the commit and go with that option.
It would also be nice if there was some way to automatically check this recipe on multiple environments, and not just the ones I have ready. How do you in general test these recipes files?